api/src/test/java/org/openmrs/hl7/handler/ORUR01HandlerTest.java

```diff
@@ -71,7 +73,7 @@
 	// hl7 parser to be used throughout
 	protected static GenericParser parser = new GenericParser();
 	
-	private static MessageTypeRouter router = new MessageTypeRouter();
+	private static ApplicationRouter router ;
 	
 	static {
 		router.registerApplication("ORU", "R01", new ORUR01Handler());
```

Comment:
please @ibacher  how can i handle this static part ?

Reply:
So I think this is a two-step process:

1. We should change `ORUR01Handler` (and `ADTA28Handler`) to extend `ReceivingApplication<Message>` instead of `Application`.

2. I would remove this `static` initializer and replace it with something more idiomatic to JUnit:

```java
@BeforeAll
public void setupApplicationRouter() {
    router = new ApplicationRouterImpl(parser);
    router.bindApplication(new AppRoutingDataImpl("ORU", "R01", "*", "*"), new ORUR01Handler());
}
```

3. Hopefully that works?

Reply:
![Screenshot from 2020-08-04 19-55-27](https://user-images.githubusercontent.com/39851246/89333359-c2c9ec00-d68c-11ea-80d2-df1b337e8aa3.png)
@ibacher  i got this from intellij.

Reply:
Ah... I did not look closely enough. Here's what I would do in `ORUR01Handler`:

1. Add a really simply method like this:

```java
/**
 * Processes an ORU R01 event message
 * 
 * This function is indented for backwards-compatibility and should not be used for new development
 *
 * @deprecated See {@link #processMessage(Message, Map)}
 */
@Deprecated
public Message processMessage(Message message) throws ReceivingApplicationException {
	return processMessage(message, new HashMap<>());
}
```

2. Modify the signature for the existing `processMessage()` implementation to this:

```java
public Message processMessage(Message message, Map<String, Object> metadata) throws ReceivingApplicationException
```

3. Inside `processMessage`, change all the `new ApplicationException()` calls to `new ReceivingApplicationException()` calls.

Reply:
done @ibacher  but the  tests fail .